### PR TITLE
Ports Sign Language from /TG/Station!

### PR DESCRIPTION
**Pull request by JackieOTheWisp**
the Radio Gloves are missing though right now, if you can fix them please do, I have it commented out so they don't clutter up the inventory and be useless :\

## About The Pull Request

Added the Tongue Tied quirk and a broken pair of Translation Gloves (which are supposed to allow you to 'speak' over comms), if you can fix this please do! I don't know honestly what I did wrong... It's been commented out of the spawn-in items meaning you don't spawn with it, so it wouldn't clutter up the inventory if you merged this like this, though I was told to save this as a draft anyways 😜 

## Why It's Good For The Game

Another quirk can always add more fun, and from an RP standpoint this could be lots of fun!

## Changelog
:cl:
add: Added Tongue Tied trait
add: Added radio gloves!
imageadd: added sprites for the gloves, and the actual tied tongue
code: changed some code, gloves still broken though
/:cl:

### CODE REVIEW

**Comment by JackieOTheWisp** [comment]
fran help what's the merge conflict mean :(

**Comment by Sinestia** [comment]
it means the file you're modifying is in a different state than what the repo has

basically the file was updated after your branch split off, likely due to this #3445 being merged

conputers try to help but it needs manually reconciled too

File: code/_globalvars/traits.dm [hunk 1]
```diff
@@ -100,7 +100,8 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_ONEWAYROAD" = TRAIT_ONEWAYROAD,
 		"TRAIT_APPRAISAL" = TRAIT_APPRAISAL,
 		"TRAIT_NOBLOCK" = TRAIT_NOBLOCK,
-		"TRAIT_NANITECOMPATIBLE" = TRAIT_NANITECOMPATIBLE
+		"TRAIT_NANITECOMPATIBLE" = TRAIT_NANITECOMPATIBLE,
+		"TRAIT_SIGN_LANG" = TRAIT_SIGN_LANG,
```
Comment (Archanial) [inline code/_globalvars/traits.dm]:
```suggestion
		"TRAIT_SIGN_LANG" = TRAIT_SIGN_LANG	
```

File: code/game/objects/items/devices/radio/radio.dm [hunk 2]
```diff
@@ -205,6 +205,20 @@
...
+			var/empty_indexes = mute.get_empty_held_indexes() //How many hands the player has empty
+			var/obj/item/clothing/gloves/radio/G = mute.get_item_by_slot(ITEM_SLOT_GLOVES)
+			if(!istype(G))
+				return FALSE
+			if(length(empty_indexes) == 1)
+				message = stars(message)
+			if(length(empty_indexes) == 0) //Due to the requirement of gloves, the arm check for normal speech would be redundant here.
+				return FALSE
+			if(mute.handcuffed)//Would be weird if they couldn't sign but their words still went over the radio
+				return FALSE
```
Comment (Archanial) [inline code/game/objects/items/devices/radio/radio.dm]:
```suggestion
	if(HAS_TRAIT(M, TRAIT_SIGN_LANG) && istype(M, /mob/living/carbon)) //Forces Sign Language users to wear the translation gloves to speak over radios
		var/mob/living/carbon/mute = M
		var/empty_indexes = mute.get_empty_held_indexes() //How many hands the player has empty
		var/gloves = mute.get_item_by_slot(ITEM_SLOT_GLOVES)
		if(!istype(gloves, /obj/item/clothing/gloves/radio))
			return FALSE
		if(length(empty_indexes) == 1)
			message = stars(message)
		if(length(empty_indexes) == 0) //Due to the requirement of gloves, the arm check for normal speech would be redundant here.
			return FALSE
		if(mute.handcuffed)//Would be weird if they couldn't sign but their words still went over the radio
			return FALSE
```

File: code/modules/mob/living/say.dm [hunk 3]
```diff
@@ -259,6 +267,25 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 		eavesdrop_range = EAVESDROP_EXTRA_RANGE
 	var/list/listening = get_hearers_in_view(message_range+eavesdrop_range, source)
 	var/list/the_dead = list()
+	if(HAS_TRAIT(src, TRAIT_SIGN_LANG))
+		var/mob/living/carbon/mute = src
+		if(istype(mute))
```
Comment (Archanial) [inline code/modules/mob/living/say.dm]:
```suggestion
		if(istype(src, /mob/living/carbon)
			var/mob/living/carbon/mute = src
```

**Comment by Effemu** [comment]
wont that make mime's vow pointless?

**Comment by Archanial** [comment]
> 
> 
> wont that make mime's vow pointless?

It won't, mimes can't use it.

**Comment by JackieOTheWisp** [comment]
gloves still don't want to work I'm testing it on my end...

**Comment by JackieOTheWisp** [comment]
...Hey archie what's those three failing tests mean? I didn't do anything to the maps...

Edit: make that 4 fails 😔

File: code/datums/traits/neutral.dm [hunk 4]
```diff
@@ -124,3 +124,30 @@
...
+//radio gloves are fuckin weird man
+
+	var/obj/item/clothing/gloves/radio/gloves = new(get_turf(H))
+	old_tongue.Remove(H)
+	new_tongue.Insert(H)
+	qdel(old_tongue)
+.H.put_in_hands(gloves)
+H.equip_to_slot(gloves, ITEM_SLOT_GLOVES)
+H.regenerate_icons()
+
```
Comment (Archanial) [inline code/datums/traits/neutral.dm]:
```suggestion
	H.put_in_hands(gloves)
	H.equip_to_slot(gloves, ITEM_SLOT_GLOVES)
	H.regenerate_icons()
```

File: code/_globalvars/traits.dm [hunk 5]
```diff
@@ -102,6 +102,7 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_NOBLOCK" = TRAIT_NOBLOCK,
 		"TRAIT_NANITECOMPATIBLE" = TRAIT_NANITECOMPATIBLE,
 		"TRAIT_WARDED" = TRAIT_WARDED
+    "TRAIT_SIGN_LANG" = TRAIT_SIGN_LANG
```
Comment (Archanial) [inline code/_globalvars/traits.dm]:
```suggestion
		"TRAIT_WARDED" = TRAIT_WARDED,
    "TRAIT_SIGN_LANG" = TRAIT_SIGN_LANG
```

**Comment by JackieOTheWisp** [comment]
Closed because of things going on with the runechat updating, I'll retry when that gets handled

File: code/modules/mob/living/say.dm [hunk 6]
```diff
@@ -241,6 +249,25 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 		eavesdrop_range = EAVESDROP_EXTRA_RANGE
 	var/list/listening = get_hearers_in_view(message_range+eavesdrop_range, source)
 	var/list/the_dead = list()
+	if(HAS_TRAIT(src, TRAIT_SIGN_LANG))
+		if(istype(src, /mob/living/carbon)
```
Comment (MNarath1) [inline code/modules/mob/living/say.dm:253]: you litterally forgot the second ) for the if the first ) is only for the istype. And yes i know i this is a closed PR

**Pull request closed without merging**